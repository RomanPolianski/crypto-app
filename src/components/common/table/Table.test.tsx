import { describe, expect, it } from 'vitest';
import { render, screen, userEvent } from '../../../../test-utils';
import { Table } from './Table';

describe('Table test', () => {
  it('is table visible', () => {
    render(
      <Table
        headers={[
          'Rank',
          'Coin',
          'Price USD',
          'Martket Cap USD',
          'Change 24hr',
          '',
        ]}
        showBody
      />
    );
    expect(screen.getByRole('table')).toBeInTheDocument();
    const welcomeText = screen.getByText(/Rank/i);
    expect(welcomeText).toBeInTheDocument();
  });

  it('Table snapshot', () => {
    const table = render(
      <Table
        headers={[
          'Rank',
          'Coin',
          'Price USD',
          'Martket Cap USD',
          'Change 24hr',
          '',
        ]}
        showBody
      />
    );
    expect(table).toMatchSnapshot();
  });
});
