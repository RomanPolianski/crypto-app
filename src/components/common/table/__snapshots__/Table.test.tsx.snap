// Vitest Snapshot v1

exports[`Table test > Table snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="_table_wv4a5_1"
      >
        <thead
          class="_table__header_wv4a5_5"
        >
          <tr
            class="_table__row_wv4a5_12"
          >
            <th
              class="_table__header__item_wv4a5_5"
            >
              Rank
            </th>
            <th
              class="_table__header__item_wv4a5_5"
            >
              Coin
            </th>
            <th
              class="_table__header__item_wv4a5_5"
            >
              Price USD
            </th>
            <th
              class="_table__header__item_wv4a5_5"
            >
              Martket Cap USD
            </th>
            <th
              class="_table__header__item_wv4a5_5"
            >
              Change 24hr
            </th>
            <th
              class="_table__header__item_wv4a5_5"
            />
          </tr>
        </thead>
        <tbody
          class="_table__body_wv4a5_12"
        />
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="_table_wv4a5_1"
    >
      <thead
        class="_table__header_wv4a5_5"
      >
        <tr
          class="_table__row_wv4a5_12"
        >
          <th
            class="_table__header__item_wv4a5_5"
          >
            Rank
          </th>
          <th
            class="_table__header__item_wv4a5_5"
          >
            Coin
          </th>
          <th
            class="_table__header__item_wv4a5_5"
          >
            Price USD
          </th>
          <th
            class="_table__header__item_wv4a5_5"
          >
            Martket Cap USD
          </th>
          <th
            class="_table__header__item_wv4a5_5"
          >
            Change 24hr
          </th>
          <th
            class="_table__header__item_wv4a5_5"
          />
        </tr>
      </thead>
      <tbody
        class="_table__body_wv4a5_12"
      />
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Table test > Take Table snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="_table_wv4a5_1"
      >
        <thead
          class="_table__header_wv4a5_5"
        >
          <tr
            class="_table__row_wv4a5_12"
          >
            <th
              class="_table__header__item_wv4a5_5"
            >
              Rank
            </th>
            <th
              class="_table__header__item_wv4a5_5"
            >
              Coin
            </th>
            <th
              class="_table__header__item_wv4a5_5"
            >
              Price USD
            </th>
            <th
              class="_table__header__item_wv4a5_5"
            >
              Martket Cap USD
            </th>
            <th
              class="_table__header__item_wv4a5_5"
            >
              Change 24hr
            </th>
            <th
              class="_table__header__item_wv4a5_5"
            />
          </tr>
        </thead>
        <tbody
          class="_table__body_wv4a5_12"
        />
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="_table_wv4a5_1"
    >
      <thead
        class="_table__header_wv4a5_5"
      >
        <tr
          class="_table__row_wv4a5_12"
        >
          <th
            class="_table__header__item_wv4a5_5"
          >
            Rank
          </th>
          <th
            class="_table__header__item_wv4a5_5"
          >
            Coin
          </th>
          <th
            class="_table__header__item_wv4a5_5"
          >
            Price USD
          </th>
          <th
            class="_table__header__item_wv4a5_5"
          >
            Martket Cap USD
          </th>
          <th
            class="_table__header__item_wv4a5_5"
          >
            Change 24hr
          </th>
          <th
            class="_table__header__item_wv4a5_5"
          />
        </tr>
      </thead>
      <tbody
        class="_table__body_wv4a5_12"
      />
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
